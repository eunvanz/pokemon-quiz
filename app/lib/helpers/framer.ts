import { Variants } from 'framer-motion'

export const motionVariants: Variants = {
  vibe: {
    translateX: [0, -4, 0, 4, 0],
    transition: {
      repeat: 2,
      duration: 0.1,
    },
  },
}
