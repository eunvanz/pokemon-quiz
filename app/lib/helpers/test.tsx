import CommonProvider from '@/app/components/common-provider'
import { PropsWithChildren } from 'react'
import api from '../api'

export interface TestProviderProps {
  api?: Partial<typeof api>
}

export const TestProvider: React.FC<PropsWithChildren<TestProviderProps>> = ({
  children,
  api = {},
}) => <CommonProvider api={api}>{children}</CommonProvider>
