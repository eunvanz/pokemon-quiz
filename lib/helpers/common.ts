export const checkIsSSR = () => {
  return typeof window === 'undefined'
}
