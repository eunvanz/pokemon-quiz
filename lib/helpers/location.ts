import { countries, zones } from 'moment-timezone/data/meta/latest.json'

export const getTimeZoneToCountryMap = () => {
  const timeZoneToCountryMap: Record<string, { name: string; abbr: string }> =
    {}
  Object.keys(zones).forEach((zone) => {
    const cityArr = zone.split('/')
    const city = cityArr[cityArr.length - 1]
    // @ts-ignore
    timeZoneToCountryMap[city] = countries[zones[zone].countries[0]]
  })
  return timeZoneToCountryMap
}

export const getUserLocationFromTimeZone = () => {
  const timeZone = Intl?.DateTimeFormat().resolvedOptions().timeZone
  if (timeZone) {
    const tzArr = timeZone.split('/')
    const region = tzArr[0]
    const city = tzArr[tzArr.length - 1]
    const { name, abbr } = getTimeZoneToCountryMap()[city]
    return {
      city,
      country: name,
      countryCode: abbr,
      region,
    }
  } else {
    return undefined
  }
}

export const getCountryFlagImgUrl = (countryCode: string) => {
  return `http://purecatamphetamine.github.io/country-flag-icons/3x2/${countryCode.toUpperCase()}.svg`
}
