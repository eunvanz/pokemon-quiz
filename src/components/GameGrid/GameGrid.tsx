import { useMemo } from "react";
import { colors } from "~/styles/colors";

export interface GameGridProps {
  width: number;
  gridItemSize: number;
}

const GameGrid: React.FC<GameGridProps> = ({ width, gridItemSize }) => {
  const gridItemCss = useMemo(() => {
    return {
      borderColor: colors.lightGray,
      borderStyle: "solid",
      borderWidth: "1px 1px 0 0",
      height: "100%",
      [`&:nth-child(${gridItemSize}n + 1)`]: {
        borderLeftWidth: "1px",
      },
      [`&:nth-child(n + ${gridItemSize ** 2 * 2 - gridItemSize + 1})`]: {
        borderBottomWidth: "1px",
      },
    };
  }, [gridItemSize]);

  return (
    <div
      css={{
        display: "grid",
        gridTemplateColumns: `repeat(${gridItemSize}, 1fr)`,
        width,
        height: width * 2,
      }}
    >
      {Array.from({ length: gridItemSize ** 2 * 2 }).map((_, idx) => (
        <div css={gridItemCss} key={idx} />
      ))}
    </div>
  );
};

export default GameGrid;
