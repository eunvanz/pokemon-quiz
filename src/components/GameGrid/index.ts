export { default } from "./GameGrid";

export type { GameGridProps } from "./GameGrid";
