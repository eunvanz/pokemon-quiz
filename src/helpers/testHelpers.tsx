import { Story } from "@storybook/react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClientProvider } from "react-query";
import api from "~/api";
import { ApiProvider } from "~/api/apiContext";
import { queryClient } from "./reactQuery";

/**
 * render storybook component in test code
 * @param StoryComponent
 * @param props
 * @returns
 */
export const renderStory = <T,>(
  StoryComponent: Story<Partial<T>>,
  props?: Partial<T>,
  renderOptions?: RenderOptions,
) => {
  return render(<StoryComponent {...props} />, renderOptions);
};

export interface TestProviderProps {
  api: Partial<typeof api>;
}

export const TestProvider: React.FC<TestProviderProps> = ({ children, api }) => (
  <ApiProvider api={api}>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </ApiProvider>
);
