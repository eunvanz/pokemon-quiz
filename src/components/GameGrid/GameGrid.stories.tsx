import type { ComponentStory, ComponentMeta } from "@storybook/react";
import { createStoryComponent } from "~/helpers/storybookHelpers";
import GameGrid from "./GameGrid";

export default {
  title: "components/GameGrid",
  component: GameGrid,
  args: {
    width: 300,
  },
} as ComponentMeta<typeof GameGrid>;

const Template: ComponentStory<typeof GameGrid> = (args) => <GameGrid {...args} />;

export const Default = createStoryComponent(Template);
