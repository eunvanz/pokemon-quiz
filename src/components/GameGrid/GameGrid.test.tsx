import { composeStories } from "@storybook/testing-react";
import { screen } from "@testing-library/react";
import { renderStory } from "~/helpers/testHelpers";
import { GameGridProps } from "./GameGrid";
import * as stories from "./GameGrid.stories";

const { Default } = composeStories(stories);

describe("GameGrid", () => {
  const setup = (props?: Partial<GameGridProps>) => renderStory(Default, props);

  it("matches with previous snapshot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  describe("gridItemSize", () => {
    it("grid items count should be [girdItemSize] ** 2 * 2", () => {
      const gridItemSize = 10;

      setup({ gridItemSize });

      const gridContainer = screen.getByTestId("game-grid-container");

      expect(gridContainer.childElementCount).toBe(200);
    });
  });
});
