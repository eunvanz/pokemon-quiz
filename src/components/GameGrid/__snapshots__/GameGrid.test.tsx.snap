// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameGrid matches with previous snapshot 1`] = `
<div>
  <div
    class="css-1nk0c6d-GameGrid"
    data-testid="game-grid-container"
  >
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
    <div
      class="css-1mc37or-GameGrid"
    />
  </div>
</div>
`;
